### ISSUE TITLE
strip passthrough (:) from combined rules functions at load time

### ISSUE THREAD

**Issue by roycewilliams** [hashcat/hashcat#3693]
When multiple rules files are being loaded and combined, any passthrough (:) rules _when combined with other rules_ either become redundant, or cancel out other functions. Removing whatever the "no op" portion of the resulting rule at load time would simplify rules parsing and analysis.

**Comment by Chick3nman** [comment]
I don't think it makes sense to strip this in hashcat. If i have 2 _different_ rule sets, i can accomplish each rule on its own PLUS the combined rules _and_ a true "pass through" (::) by having passthrough in both rules. If you don't want : in the rule set, I think it's on the user to remove it intentionally.

**Comment by roycewilliams** [comment]
It's the *product* of the rules I'm talking about - rules that end up
looking like this:

`: : : : $1 $2 $3` 

... because they were *combined* dynamically with other sets of rules (when stacking them using hashcat itself)

**Comment by Chick3nman** [comment]
So you want `: : $1 $2` to become `$1 $2` in memory?

**Comment by roycewilliams** [comment]
Yep, exactly  (and I assume also not written to debug output as well).

**Comment by Chick3nman** [comment]
Ahh, this would help with "function limits" but it wouldn't dedupe `: $1` and `$1 :`. Though I've been meaning to look at doing rule dedupe for "literally identical"...
